lib/state/simperium/middleware.ts

```diff
@@ -251,7 +251,8 @@ export const initSimperium = (
           action.meta?.nextNoteToOpen ??
           getState().ui.openedNote;
 
-        if (noteId) {
+        //  Preload the revisions when opening a note but only do it if no revisions are in memory
+        if (noteId && !nextState.data.noteRevisions.get(noteId)?.size) {
```

Comment:
we might need to add a new flag saying "we're getting them" otherwise we might issue a bunch of requests in the time before the first one comes back.

something like a set in this middleware could do.

```ts
const noteQueue = new BucketQueue(noteBucket);
const queueNoteUpdate = …

const hasRequestedRevisions = new Set<T.EntityId>();
```

then when checking here…

```ts
if (noteId && !nextState.data.noteRevisions.get(noteId)?.size && !hasRequestedRevisions.has(noteId)) {
	hasRequestedRevisions.add(noteId);
	setTimeout( … );
	…
}
```